src/lfview/lf-data-provider.ts

```diff
@@ -0,0 +1,487 @@
+            });
+        });
+    }
+
+    /**
+     * Imports the reactor associated with the selected LFDataProviderNode.
+     * @param node - The node representing the reactor to import.
+     */
+    async importReactorCommand(node: LFDataProviderNode) {
+        const editor = vscode.window.activeTextEditor;
+        if (editor) {
+            if (!editor.document.uri.fsPath.endsWith('.lf')) {
+                vscode.window.showErrorMessage('The active editor must be a Ligua Franca program.');
+                return;
+            }
+            const relativePath = this.getRelativePath(editor.document.uri.fsPath, node.uri.fsPath);
+            const importText = `import ${node.label.toString()} from "${relativePath}"\n`;
+            const position = await this.getTargetPosition(editor.document.uri);
+            this.addTextOnActiveEditor(editor, position.end, importText);
+        }
+    }
+
+    /**
+     * Gets the selection to highlight in the editor for the node.
+     * @param node - The node for which to get the selection.
+     * @returns The selection to highlight.
+     */
+    getHighlightSelection(node: LFDataProviderNode): vscode.Selection {
+        const editor = vscode.window.activeTextEditor;
+        if (editor) {
+            const sel = new vscode.Selection(node.position.start - 1, 0, node.position.start - 1, this.HIGHLIGHT_OFFSET);
+            const selectionRange = new vscode.Range(sel.start.line, sel.start.character, sel.end.line, sel.end.character);
+            const highlighted = editor.document.getText(selectionRange);
+            const idx = highlighted.indexOf(node.label.toString());
+            const endIdx = idx + node.label.toString().length;
+            return new vscode.Selection(node.position.start - 1, idx, node.position.start - 1, endIdx);
+        }
+        return new vscode.Selection(node.position.start - 1, 0, node.position.start - 1, this.HIGHLIGHT_OFFSET);
+    }
+
+    /**
+     * Gets the relative path between a source and target file.
+     * @param source The path to the source file.
+     * @param target The path to the target file.
+     * @returns The relative path between the source and target files.
+     */
+    getRelativePath(source: string, target: string) {
+        return path.relative(path.dirname(source), target);
+    }
+
+    /**
+     * Adds text to the active editor at the specified position.
+     * @param editor - The active text editor.
+     * @param end - The position to start searching for the next empty line from.
+     * @param importText - The text to insert.
+     */
+    addTextOnActiveEditor(editor: vscode.TextEditor, end: number, importText: string): void {
+        const document = editor.document;
+        const totalLines = document.lineCount;
+
+        let startIndex = end;
+        let foundEmptyLine = false;
+
+        while (startIndex < totalLines) {
+            const line = document.lineAt(startIndex);
+            if (line.isEmptyOrWhitespace) {
```

Comment:
This is not robust. What if this empty line is in the middle of a multiline comment? I think that it is a code smell to perform an automatic modification on a raw string of code without parsing it properly. It would probably be better to run the parser on the current file and find the correct edit location based on the concrete syntax tree, which in the XText world is the node model (you can grep our code base for `NodeModelUtils`).

Reply:
Thank you for your suggestions, Peter. I was looking for something like that. I will take a look at NodeModeUtils to improve this function.